### PR TITLE
feat(ci): enable Github Actions for external PRs and master

### LINKED ISSUE #1471: CI builds of third-party branches time out and fail
All of our CI builds of third-party branches time out and fail. ([Example](https://buildkite.com/monadic/radicle-upstream/builds/6936#b81ce7d7-bc85-401a-8e56-7279f8e7f24c/157-265))

They hang at the following line:
~~~
test can_ask_and_clone_project ... fatal: '.' does not appear to be a git repository
~~~
Then they time out after 60 minutes.

This is probably due to us using the `kata` container runtime for third-party builds.

### PR DESCRIPTION
**Pull request by geigerzaehler**
To fix #1471 we enable Github actions for external pull requests. We also build the master branch on Github Actions.

### CODE REVIEW

**Comment by geigerzaehler** [comment]
Unfortunately, the rust tests are still flaky. I’m waiting for #1464 to see if this improves things.

**Comment by FintanH** [comment]
> Unfortunately, the rust tests are still flaky. I’m waiting for #1464 to see if this improves things.

Wut, why do we need #1464 for thing to improve? :eyes:

**Comment by geigerzaehler** [comment]
> > Unfortunately, the rust tests are still flaky. I’m waiting for #1464 to see if this improves things.
> 
> Wut, why do we need #1464 for thing to improve? eyes

Oops, wrong PR. It’s #1463.

**Comment by geigerzaehler** [comment]
This is ready and working now.

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: build
+on:
+  push:
+    branches: [master]
+  pull_request:
+jobs:
+  build-linux:
+    # If this was triggered by a pull request only run it if it
```
Comment (NunoAlexandre) [inline .github/workflows/build.yaml:8]:
```suggestion
    # If this was triggered by a pull request, only run it if it
```

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -448,21 +448,13 @@ Upstream.
...
-All relevant configuration can be found here:
-
-```sh
-radicle-upstream/.buildkite
-.
-├── Dockerfile
-├── pipeline.yaml
-└── run.sh
-```
+We run CI build with [Buildkite][bk] [Github Actions][ga]. Buildkite builds all
```
Comment (NunoAlexandre) [inline DEVELOPMENT.md:451]:
```suggestion
We run CI build with [Buildkite][bk] and [Github Actions][ga]. Buildkite builds all
```

**Review by NunoAlexandre — COMMENTED** [review]
Amazing :speedboat:

File: DEVELOPMENT.md (hunk 2 above)
Comment (rudolfs) [inline DEVELOPMENT.md]:
```suggestion
We run CI builds with [Buildkite][bk] and [Github Actions][ga]. Buildkite builds all
```

**Review by rudolfs — APPROVED** [review]

**Review by rudolfs — DISMISSED** [review]

**Review by NunoAlexandre — APPROVED** [review]

**Pull request merged**